src/network/netmon.c

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2023 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include "sntp_client.h"
+#include "dhcpwait.h"
+#include "netmon_stats.h"
+
+LOG_MODULE_REGISTER(netmon, CONFIG_LOG_DEFAULT_LEVEL);
+
+#define NETMON_THREAD_PRIORITY 5
+static struct k_work_delayable time_sync;
+
+/* Network monitor worker thread work item to sync
+ * the time with SNTP server.
+ */
+static void netmon_time_sync(struct k_work *work)
+{
+
+	if (!is_dhcp_up()) {
+		LOG_ERR("Network down");
+		goto end;
+	}
+	sntp_resync_time();
+	netmon_sntp_time_sync_call_stats(get_netmon_stats_inst());
+
+	/* Reschedule the worker therad to resync the time base with SNTP server. */
+end:
+	k_work_reschedule(&time_sync, K_HOURS(24));
```

Comment:
Any concerns that this will drift as it will generally be slightly more than 24 hours?